src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocSummaryLocationCheck.java

```diff
@@ -0,0 +1,275 @@
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties" file.
+     */
+    public static final String MSG_JAVADOC_SUMMARY_FIRST_LINE = "javadoc.summary.first.line";
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties" file.
+     */
+    public static final String MSG_JAVADOC_SUMMARY_SECOND_LINE = "javadoc.summary.second.line";
+
+    /**
+     * The pattern for detecting text ignored by the javadoc tool.
+     */
+    private static final Pattern OPTIONAL_PREFIX = Pattern.compile("^(\\**\\s*){1,2}?$");
+
+    /**
+     * Specify the policy on placement of the javadoc summary.
+     */
+    private JavadocSummaryLocation location = JavadocSummaryLocation.SECOND_LINE;
+
+    @Override
+    public int[] getDefaultJavadocTokens() {
+        return new int[] {
+            JavadocTokenTypes.JAVADOC,
+        };
+    }
+
+    @Override
+    public int[] getRequiredJavadocTokens() {
+        return getAcceptableJavadocTokens();
+    }
+
+    @Override
+    public void visitJavadocToken(DetailNode ast) {
+        if (isMultilineComment(ast)) {
+            final JavadocSummaryLocation javadocSummaryLocation = getJavadocSummaryLocation(ast);
+            if (javadocSummaryLocation != location) {
+                if (location == JavadocSummaryLocation.FIRST_LINE) {
+                    log(ast.getLineNumber(), MSG_JAVADOC_SUMMARY_FIRST_LINE);
+                }
+                else {
+                    log(ast.getLineNumber(), MSG_JAVADOC_SUMMARY_SECOND_LINE);
+                }
+            }
+        }
+    }
+
+    /**
+     * Setter to specify the policy on placement of the javadoc summary.
+     * @param value string to decode location from
+     * @throws IllegalArgumentException if unable to decode
+     */
+    public void setLocation(String value) {
+        location = JavadocSummaryLocation.valueOf(value.trim().toUpperCase(Locale.ENGLISH));
+    }
+
+    /**
+     * Checks if a DetailNode of type {@code JavadocTokenTypes.JAVADOC} has
+     * more than one line.
+     *
+     * @param node node to check
+     * @return {@code true} for
+     */
+    private static boolean isMultilineComment(DetailNode node) {
+        final DetailNode[] children = node.getChildren();
+        final int firstChildLineNo = children[0].getLineNumber();
+        final int lastChildLineNo = children[children.length - 1].getLineNumber();
+        return firstChildLineNo != lastChildLineNo;
+    }
+
+    /**
+     * Returns the javadoc summary location for the {@code JavadocTokenTypes.JAVADOC}
+     * node ignoring any leading asterisks and whitespace.
+     *
+     * @param node node to return
+     * @return {@link JavadocSummaryLocation#FIRST_LINE} for scala-style comments;
+     *         {@link JavadocSummaryLocation#SECOND_LINE} for java-style comments.
+     */
+    private static JavadocSummaryLocation getJavadocSummaryLocation(DetailNode node) {
```

Comment:
What is the summary? If I have an example like:
````
/** <p>
* Text.
* </p>
 */
````
Isn't the word `Text` the summary, so this wouldn't cause any violations for SECOND, but will for FIRST?

Reply:
I think the text starts in the first line in this case. Added this to test cases.